src/main/java/org/folio/consortia/service/impl/SharingRoleService.java

```diff
@@ -0,0 +1,131 @@
+package org.folio.consortia.service.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.ObjectUtils;
+import org.folio.consortia.domain.dto.PublicationRequest;
+import org.folio.consortia.domain.dto.SharingRoleDeleteResponse;
+import org.folio.consortia.domain.dto.SharingRoleRequest;
+import org.folio.consortia.domain.dto.SharingRoleResponse;
+import org.folio.consortia.domain.entity.SharingRoleEntity;
+import org.folio.consortia.exception.ResourceNotFoundException;
+import org.folio.consortia.repository.SharingRoleRepository;
+import org.folio.consortia.service.BaseSharingService;
+import org.folio.consortia.service.ConsortiumService;
+import org.folio.consortia.service.PublicationService;
+import org.folio.consortia.service.TenantService;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+
+@Service
+@Log4j2
+public class SharingRoleService extends BaseSharingService<SharingRoleRequest, SharingRoleResponse, SharingRoleDeleteResponse, SharingRoleEntity> {
+
+  private static final String TYPE = "type";
+
+  private final SharingRoleRepository sharingRoleRepository;
+  private final ObjectMapper objectMapper;
+
+  public SharingRoleService(TenantService tenantService, ConsortiumService consortiumService,
+                            SystemUserScopedExecutionService systemUserScopedExecutionService,
+                            PublicationService publicationService, FolioExecutionContext folioExecutionContext,
+                            ObjectMapper parentObjectMapper, TaskExecutor asyncTaskExecutor, SharingRoleRepository sharingRoleRepository, ObjectMapper objectMapper) {
+    super(tenantService, consortiumService, systemUserScopedExecutionService, publicationService,
+      folioExecutionContext, parentObjectMapper, asyncTaskExecutor);
+    this.sharingRoleRepository = sharingRoleRepository;
```

Comment:
Member field constructor setters for `sharingRoleRepository` & `objectMapper` can probably be replaced with `@RequiredArgsConstructor` annotation